packages/mui-joy/src/TextField/TextFieldProps.ts

```diff
@@ -0,0 +1,77 @@
+import * as React from 'react';
+import { OverrideProps } from '@mui/types';
+import { FormHelperTextProps } from '../FormHelperText/FormHelperTextProps';
+import { FormLabelProps } from '../FormLabel/FormLabelProps';
+import { InputProps } from '../Input/InputProps';
+
+export type TextFieldSlot = 'root';
+
+export interface TextFieldTypeMap<P = {}, D extends React.ElementType = 'div'> {
+  props: P &
+    Pick<
+      InputProps,
+      | 'autoComplete'
+      | 'autoFocus'
+      | 'disabled'
+      | 'error'
+      | 'required'
+      | 'fullWidth'
+      | 'inputRef'
+      | 'placeholder'
+      | 'defaultValue'
+      | 'value'
+      | 'onChange'
+      | 'onFocus'
+      | 'onBlur'
+      | 'type'
+      | 'variant'
+      | 'color'
+      | 'size'
+      | 'startAdornment'
+      | 'endAdornment'
+    > & {
+      components?: {
+        Root?: React.ElementType;
+        Label?: React.ElementType;
+        HelperText?: React.ElementType;
+        InputRoot?: React.ElementType;
+        InputInput?: React.ElementType;
+      };
+      componentsProps?: {
+        root?: React.ComponentPropsWithRef<'div'>;
+        label?: FormLabelProps;
+        helperText?: FormHelperTextProps;
+        inputRoot?: React.ComponentPropsWithRef<'div'>;
+        inputInput?: React.ComponentPropsWithRef<'input'>;
+      };
```

Comment:
@michaldudak Is this resonates with the way you thought? The case of `TextField` is special because it contains `Input` component and the `Input` has `input` as HTML, so I have to use `inputInput`.

Reply:
I'm leaning towards composing such components from lower-level components, not primitive DOM elements. So instead of having `InputRoot` and `InputInput`, I'd put `Input` that would default to Joy's Input component. 
This way, more abstract composite components won't have a long list of slots to customize. Plus, it allows developers to provide their own Inputs with custom logic if necessary.

Reply:
Got it, that's a lot better.